.github/workflows/test.yaml

```diff
@@ -0,0 +1,86 @@
+name: Test
+
+on:
+  # Run CI on pushes to the main branch, and on PRs against main.
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  PUB_ENVIRONMENT: bot.github
+  _PUB_TEST_SNAPSHOT: ${{ github.workspace }}/.dart_tool/pub.dart.snapshot.dart2
+  _TESTS_FILE: .dart_tool/test_files
+
+jobs:
+  # Check code formatting and static analysis on a single OS (linux)
+  # against Dart stable and dev.
+  analyze:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        sdk: [dev]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache Pub hosted dependencies
+        uses: actions/cache@v2
+        with:
+          path: "~/.pub-cache/hosted"
+          key: "os:ubuntu-latest;pub-cache-hosted;dart:${{ matrix.sdk }};packages:pub"
+          restore-keys: |
+            os:ubuntu-latest;pub-cache-hosted;dart:${{ matrix.sdk }};packages:pub
+            os:ubuntu-latest;pub-cache-hosted;dart:${{ matrix.sdk }}
+            os:ubuntu-latest;pub-cache-hosted
+            os:ubuntu-latest
+      - uses: dart-lang/setup-dart@v0.1
+        with:
+          channel: ${{ matrix.sdk }}
+      - name: Install dependencies
+        run: dart pub get
+      - name: Check formatting
+        run: dart format --output=none --set-exit-if-changed .
+      - name: Analyze code
+        run: dart analyze
+
+  # Run tests on a matrix consisting of three dimensions:
+  # 1. OS: mac, windows, linux
+  # 2. release channel: dev
+  # 3. shard: 0, 1, 2, 3, 4, 5, 6
+  test:
+    needs: analyze
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        sdk: [dev]
+        shard: [0, 1, 2, 3, 4, 5, 6]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache Pub hosted dependencies
+        uses: actions/cache@v2
+        with:
+          path: "~/.pub-cache/hosted"
+          key: "os:ubuntu-latest;pub-cache-hosted;dart:${{ matrix.sdk }};packages:pub"
+          restore-keys: |
+            os:${{ matrix.os }};pub-cache-hosted;dart:${{ matrix.sdk }};packages:pub
```

Comment:
I don't really understand these lines.
Could you add a comment what they do?

Reply:
They are a prefix search for less specific cache keys, it makes some sense in mono repos but not really here. Anyway, I deleted the whole cache step.